tests/python_package_test/test_engine.py

```diff
@@ -995,45 +995,68 @@ def test_init_with_subset(self):
         self.assertEqual(subset_data_3.get_data(), "lgb_train_data.bin")
         self.assertEqual(subset_data_4.get_data(), "lgb_train_data.bin")
 
-    def test_monotone_constraint(self):
+    # test if categorical features and monotone features can both be in a dataset without causing issues
```

Comment:
Is it `TODO` comment?

Reply:
Yes, I forgot about this one sorry. I am not sure anymore if it is worth having a test for that. I actually think that's fine now, I was more worried about that with my old implementation which was much worse. If you want me to add a python test for that I can do it but otherwise, I think just deleting the comment is fine. Let me know what you think. Thanks

Reply:
I believe that having a simple (not so complicated as the current one, just to be sure it will not crash) test for cat. features + constraints will be better than not having a such test. Maybe it is not so useful for now as you feel confident with the current implementation. But it will help spot issues with possible future refactoring or new constraint methods.

Reply:
I updated the current test that checks for monotonicity to include a categorical variable. That way if someone wants to improve the handling of categorical variables with monotonic constraints, the test will already be there. Let me know if you are happy with that.